packages/git/src/browser/git-quick-open-service.ts

```diff
@@ -38,9 +39,55 @@ export class GitQuickOpenService {
         @inject(Git) protected readonly git: Git,
         @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider,
         @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
-        @inject(MessageService) protected readonly messageService: MessageService
+        @inject(MessageService) protected readonly messageService: MessageService,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
     ) { }
 
+    async clone(url?: string, folder?: string, branch?: string): Promise<void> {
+        if (!folder) {
+            const roots = await this.workspaceService.roots;
+            folder = roots[0].uri;
+        }
+
+        const __folder = folder;
+        if (url) {
+            return this.git.clone(url, { localUri: this.buildDefaultProjectPath(folder, url), branch: branch }).then(() => { });
```

Comment:
You can just do:
```js
return await this.git.clone(url, { localUri: this.buildDefaultProjectPath(folder, url), branch });
```

Reply:
I have refactored a bit to return the repo string. It would be useful when using Git clone command from API.